leader/leader.go

```diff
@@ -175,51 +185,51 @@ func (l *Client) AddWatch(key string, retry time.Duration, valuesC chan string)
 				}
 			}
 
-			if err != nil {
-				select {
-				case <-ticker.C:
-					steps += 1
+			// The watcher has received a new event, send it to the client
+			// unless this value has already been sent.
+			if err == nil {
+				if resp.PrevNode != nil && resp.PrevNode.Value == resp.Node.Value && sentAnything {
```

Comment:
Doesn't this have the possibility to nil deref? On the first loop err and watcher will be nil, if the watcher doesn't get created because the client is closing, err and resp will be nil which will then try and process the resp.